sa/sa.go

```diff
@@ -167,23 +167,28 @@ func (ssa *SQLStorageAuthority) GetRegistrationByKey(ctx context.Context, key *j
 }
 
 // GetAuthorization obtains an Authorization by ID
-func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string) (authz core.Authorization, err error) {
+func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string) (core.Authorization, error) {
+	authz := core.Authorization{}
 	tx, err := ssa.dbMap.Begin()
 	if err != nil {
-		return
+		return authz, err
 	}
-
 	pa, err := selectPendingAuthz(tx, "WHERE id = ?", id)
 	if err != nil && err != sql.ErrNoRows {
-		err = Rollback(tx, err)
-		return
+		return authz, Rollback(tx, err)
 	}
 	if err == sql.ErrNoRows {
 		var fa authzModel
-		err = tx.SelectOne(&fa, fmt.Sprintf("SELECT %s FROM authz WHERE id = ?", authzFields), id)
-		if err != nil {
-			err = Rollback(tx, err)
-			return
+		err := tx.SelectOne(&fa, fmt.Sprintf("SELECT %s FROM authz WHERE id = ?", authzFields), id)
+		if err != nil && err != sql.ErrNoRows {
+			return authz, Rollback(tx, err)
+		} else if err == sql.ErrNoRows {
+			// Since we have not done any work at this point we can commit the
+			// transaction instead of rolling it back
+			if rollbackErr := tx.Commit(); rollbackErr != nil {
+				return authz, rollbackErr
+			}
+			return authz, err
```

Comment:
Here we should be returning a `berrors.NotFound` and testing for it in the respective WFE handler that calls this.